graph/src/schema/input_schema.rs

```diff
@@ -0,0 +1,357 @@
+use std::collections::{BTreeMap, HashSet};
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Error};
+use store::Entity;
+
+use crate::cheap_clone::CheapClone;
+use crate::components::store::{EntityKey, EntityType, LoadRelatedRequest};
+use crate::data::graphql::ext::DirectiveFinder;
+use crate::data::graphql::{DirectiveExt, DocumentExt, ObjectTypeExt, TypeExt, ValueExt};
+use crate::data::store::{self, scalar, IntoEntityIterator, TryIntoEntityIterator};
+use crate::prelude::q::Value;
+use crate::prelude::{s, DeploymentHash};
+use crate::schema::api_schema;
+use crate::util::intern::AtomPool;
+
+use super::fulltext::FulltextDefinition;
+use super::{ApiSchema, Schema, SchemaValidationError};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct InputSchema {
+    inner: Arc<Inner>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Inner {
```

Comment:
Does it make sense for Inner to be pub?

Reply:
I removed the `impl Inner` and the `Deref`